File: python/src/sorting.py
class Sorting:
    def bubble_sort(self, items):
        length = len(items)
        for i in range(length, 0, -1):
            no_swaps = True
            for j in range(i - 1):
                if items[j] > items[j + 1]:
                    items[j], items[j + 1] = items[j + 1], items[j]
                    no_swaps = False
            if no_swaps:
                break
        return items

    def second_largest_element(self, array):
        largest_element = max(array)
        element = -float('inf')
        for i in array:
            if element < i and i != largest_element:
                element = i
        return element

File: python/src/sorting.js
class Sorting {
    // https://the-algorithms.com/algorithm/bubble-sort?lang=javascript
    bubbleSort(items) {
        const length = items.length
        let noSwaps

        for (let i = length; i > 0; i--) {
            // flag for optimization
            noSwaps = true
            // Number of passes
            for (let j = 0; j < i - 1; j++) {
                // Compare the adjacent positions
                if (items[j] > items[j + 1]) {
                    // Swap the numbers
                    ;[items[j], items[j + 1]] = [items[j + 1], items[j]]
                    noSwaps = false
                }
            }
            if (noSwaps) {
                break
            }
        }

        return items
    }

    // https://the-algorithms.com/algorithm/find-second-largest-element?lang=javascript
    secondLargestElement = (array) => {
        const largestElement = Math.max(...array)
        let element = -Number.MAX_VALUE
      
        for (let i = 0; i < array.length; i++) {
          if (element < array[i] && array[i] !== largestElement) {
            element = array[i]
          }
        }
      
        return element
      }
    
}

export default Sorting;
